include/swift/AST/Requirement.h

```diff
@@ -43,8 +46,11 @@ enum class RequirementKind : unsigned {
 /// \brief A single requirement placed on the type parameters (or associated
 /// types thereof) of a
 class Requirement {
-  llvm::PointerIntPair<Type, 2, RequirementKind> FirstTypeAndKind;
-  Type SecondType;
+  llvm::PointerIntPair<Type, 3, RequirementKind> FirstTypeAndKind;
+  union {
```

Comment:
And if one of those new alternatives will not be a pointer-like something, we won't be able to use union anymore. 

So, do you still think we should go for a PointerUnion?